internal/provider/resource_aws_transit_gateway_attachment.go

```diff
@@ -0,0 +1,316 @@
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"provider_transit_gateway_attachment_id": {
+				Description: "The Transit gateway attachment ID used by AWS.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"state": {
+				Description: "The state of the Transit gateway attachment.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time that the Transit gateway attachment was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"expires_at": {
+				Description: "The time after which the Transit gateway attachment will be considered expired if it hasn't transitioned into 'Accepted' or 'Active' state.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceAwsTransitGatewayAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	hvnID := d.Get("hvn_id").(string)
+	tgwAttachmentID := d.Get("transit_gateway_attachment_id").(string)
+	tgwID := d.Get("transit_gateway_id").(string)
+	resourceShareARN := d.Get("resource_share_arn").(string)
+	rawCIDRs := d.Get("destination_cidrs").([]interface{})
+
+	destinationCIDRs := make([]string, len(rawCIDRs))
+	for i, cidr := range rawCIDRs {
+		strCidr, ok := cidr.(string)
+		if !ok {
+			return diag.Errorf("unable to convert cidr: %v to string", cidr)
+		}
+		destinationCIDRs[i] = strCidr
+	}
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	// Check for an existing HVN
+	_, err := clients.GetHvnByID(ctx, client, loc, hvnID)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to find the HVN (%s) for the Transit gateway attachment", hvnID)
+		}
+
+		return diag.Errorf("unable to check for presence of an existing HVN (%s): %v", hvnID, err)
+	}
+	log.Printf("[INFO] HVN (%s) found, proceeding with create", hvnID)
+
+	// Check if TGW attachment already exists
+	_, err = clients.GetTGWAttachmentByID(ctx, client, tgwAttachmentID, hvnID, loc)
+	if err != nil {
+		if !clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to check for presence of an existing Transit gateway attachment (%s): %v", tgwAttachmentID, err)
+		}
+
+		log.Printf("[INFO] Transit gateway attachment (%s) not found, proceeding with create", tgwAttachmentID)
+	} else {
+		return diag.Errorf("a Transit gateway attachment with transit_gateway_attachment_id=%s, hvn_id=%s and project_id=%s already exists - to be managed via Terraform this resource needs to be imported into the state. Please see the resource documentation for hcp_transit_gateway_attachment for more information", tgwAttachmentID, hvnID, loc.ProjectID)
```

Comment:
Hmm not sure if this is the pattern we follow everywhere so please ignore this comment if this is a custom thing, but [a check ensuring unique tgwAttachmentID within a project is in place](https://github.com/hashicorp/cloud-network/blob/master/service/network/private/service_create_tgw_attachment.go#L341), and would be part of the output in the subsequent `Create` request.

Reply:
Ah yea, this came up with the HVN resource, and we ended up with the pattern you see here where we do a GET first: https://github.com/hashicorp/terraform-provider-hcp/pull/3#discussion_r556234766

We might be able to accomplish the same thing with the CREATE call, but I think we would need to be careful about what guarantees we have from the endpoint (eg. it looks like we can get at least two types of `AlreadyExists` response codes), and the main driver here was being able to supply an actionable error message.

Reply:
Thanks for explaining, this sounds good to me, resolving this comment